hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/SimpleServerRpcConnection.java

```diff
@@ -252,13 +253,14 @@ private void saslReadAndProcess(ByteBuff saslToken) throws IOException, Interrup
         doRawSaslReply(SaslStatus.SUCCESS, new BytesWritable(replyToken), null, null);
       }
       if (saslServer.isComplete()) {
-        String qop = saslServer.getNegotiatedQop();
-        useWrap = qop != null && !"auth".equalsIgnoreCase(qop);
+        String negotiatedQop = saslServer.getNegotiatedQop();
```

Comment:
So in the old code, SimpleServerRpcConnection does not call finishSaslNegotiation? Seems a bit strange...

Reply:
That's a good point.
finishSaslNegotiation does the same thing, but takes care of logging and metrics, it's better to call that from here.

Reply:
Refactored SimpleServerRpcConnection to use finishSaslNegotiation.